src/BucketTesting/Logging/DatabaseBucketLogger.php

```diff
@@ -40,4 +40,10 @@ private function addBucketLogBuckets( BucketLog $bucketLog, array $buckets ): vo
 			);
 		}
 	}
+
+	private function campaignIsNotRunning( Campaign $campaign ): bool
+	{
+		$now = new CampaignDate();
+		return $campaign->isActive() && !$campaign->isExpired( $now );
```

Comment:
```suggestion
		return !$campaign->isActive() || $campaign->isExpired( $now );
```
not sure if it was even correct before, but this here might be easier to understand?
the campaign being not active (->true) leads to campaign not running
or
the campaign being expired (->) true leads to campaign not running

Reply:
You're right. Originally, I named this `campaignIsActive` and inverted your condition. But then I had the negation on the call, so i renamed it campaignIsNotActive and forgot to invert the condition again. Luckily, your unit test caught this!